Nos/Service/NostrMetadata/TLVEntity.swift

```diff
@@ -0,0 +1,82 @@
+import Foundation
+
+/// A TLV (type-length-value) entity.
+/// - Note: See [NIP-19](https://github.com/nostr-protocol/nips/blob/master/19.md) for more information.
+enum TLVEntity {
+    /// The special type with its associated value as a String.
+    case special(value: String)
+    
+    /// The relay type with its associated value as a String.
+    case relay(value: String)
+
+    /// The author type with its associated value as a String.
+    case author(value: String)
+    
+    /// The kind type with its associated value as a UInt32.
+    case kind(value: UInt32)
+
+    /// An unknown type.
+    case unknown
+    
+    /// Decodes the given binary-encoded list of TLV (type-length-value). The meaning of the decoded value will depend
+    /// on the Bech32 human readable part. Unreadable entities are included in the array as `.unknown`.
+    /// - Parameter data: The TLV data to decode.
+    /// - Returns: An array containing decoded TLV entities, with `.unknown` values for any that could not be decoded.
+    static func decodeEntities(data: Data) -> [TLVEntity] {
+        guard let converted = try? data.base8FromBase5() else {
+            return []
+        }
+
+        var result: [TLVEntity] = []
+
+        var offset = 0
+        while offset + 1 < converted.count {
+            let rawType = converted[offset]
+            let type = TLVType(rawValue: rawType)
+            let length = Int(converted[offset + 1])
+            let value = converted.subdata(in: offset + 2 ..< offset + 2 + length)
+
+            let entity: TLVEntity
+            switch type {
+            case .special:
+                let valueString = SHA256Key.decode(base8: value)
+                entity = .special(value: valueString)
+            case .relay:
+                if let valueString = String(data: value, encoding: .ascii) {
```

Comment:
This would be better described with a `guard`

Reply:
Agreed. Unfortunately, guard doesn't work here since we're not returning, breaking, continuing, or throwing. (I think those are all the options.) Unless you're suggesting we don't use the `entity = .unknown` line and instead continue the loop... That would work, but I think I'd like to have the `.unknown` in the array just in case we do have trouble decoding a relay so we have an indication that something went wrong on one of the entities.